test/unit-test/test_lib/test_analysis/test_user.py

```diff
@@ -1,34 +1,146 @@
+        expected_user_words_dict = util.load_from_disk(current_directory + "/data/user_test/keywords/user_words_dict")
+        expected_nicks_for_stop_words = util.load_from_disk(current_directory + "/data/user_test/keywords/nicks_for_stop_words")
+        expected_sorted_keywords_for_channels = util.load_from_disk(current_directory + "/data/user_test/keywords/sorted_keywords_for_channels")
+
+        keywords_filtered, user_keyword_freq_dict, user_words_dict, nicks_for_stop_words, sorted_keywords_for_channels = user.keywords(self.log_data , self.nicks, self.nick_same_list)
+
+        self.assertEqual(expected_keywords_filtered, keywords_filtered)
+        self.assertEqual(expected_user_keyword_freq_dict, user_keyword_freq_dict)
+        self.assertEqual(expected_user_words_dict, user_words_dict)
+        self.assertEqual(expected_nicks_for_stop_words, nicks_for_stop_words)
+        self.assertEqual(expected_sorted_keywords_for_channels, sorted_keywords_for_channels)
+
+
+    @patch("lib.analysis.user.keywords", autospec = True)
+    @patch("lib.analysis.user.extended_stop_words", autospec = True)
+    @patch("lib.analysis.user.time", autospec = True)
+    def test_keywords_clusters(self, mock_time, mock_extended_stop_words, mock_keywords):
+        keywords_filtered = util.load_from_disk(current_directory + "/data/user_test/keywords/keywords_filtered")
+        user_keyword_freq_dict = util.load_from_disk(current_directory + "/data/user_test/user_keyword_freq_dict")
+        user_words_dict = util.load_from_disk(current_directory + "/data/user_test/keywords/user_words_dict")
+        nicks_for_stop_words = util.load_from_disk(current_directory + "/data/user_test/keywords/nicks_for_stop_words")
+        sorted_keywords_for_channels = util.load_from_disk(current_directory + "/data/user_test/keywords/sorted_keywords_for_channels")
+
+        mock_keywords.return_value = keywords_filtered, user_keyword_freq_dict, user_words_dict, nicks_for_stop_words, sorted_keywords_for_channels
+        mock_extended_stop_words.return_value = util.load_from_disk(current_directory + "/data/user_test/extended_stop_words")
+        mock_time.return_value = 0
+        expected_captured_output = util.load_from_disk( current_directory + "/data/user_test/stdout_captured_output_keywords_clusters");
+
+        captured_output = StringIO.StringIO()
+        sys.stdout = captured_output
+        user.keywords_clusters(self.log_data, self.nicks, self.nick_same_list, current_directory + "/data/user_test/","temp_output_keywords_clusters")
+        sys.stdout = sys.__stdin__
+        output = captured_output.getvalue()
+
+        self.assertEqual(expected_captured_output, output)
+        self.assertTrue(filecmp.cmp(current_directory + "/data/user_test/output_keywords_clusters.txt", current_directory + "/data/user_test/temp_output_keywords_clusters.txt"))
```

Comment:
@Kausam Based on our preferences, we are going towards reading the file into memory and asserting on the equality of the contents.